lib/datadog/tracing/contrib/opensearch/patcher.rb

```diff
@@ -0,0 +1,116 @@
+            def perform_request(method, path, params = {}, body = nil, headers = nil)
+              response = nil
+              Tracing.trace('opensearch.query', service: datadog_configuration[:service_name]) do |span|
+                begin
+                  # Set generic tags
+                  span.set_tag(Tracing::Metadata::Ext::TAG_COMPONENT, Ext::TAG_COMPONENT)
+                  span.set_tag(Tracing::Metadata::Ext::TAG_KIND, Tracing::Metadata::Ext::SpanKind::TAG_CLIENT)
+                  span.set_tag(Contrib::Ext::DB::TAG_SYSTEM, Ext::TAG_SYSTEM)
+
+                  # Set argument tags
+                  span.set_tag(OpenSearch::Ext::TAG_METHOD, method)
+                  span.set_tag(OpenSearch::Ext::TAG_PATH, path)
+
+                  tag_params(params, span)
+                  tag_body(body, span)
+
+                  # Parse url
+                  original_url = transport.get_connection.full_url(path, {})
+                  url = URI.parse(original_url)
+                  host = url.host
+                  port = url.port
+                  scheme = url.scheme
+                  # Set url.user to nil to remove sensitive information (i.e. user's username and password)
+                  url.user = nil
+
+                  # Set url tags
+                  span.set_tag(OpenSearch::Ext::TAG_URL, url)
+                  span.set_tag(OpenSearch::Ext::TAG_HOST, host)
+                  span.set_tag(OpenSearch::Ext::TAG_PORT, port)
+                  span.set_tag(OpenSearch::Ext::TAG_SCHEME, scheme)
+
+                  # Define span resource
+                  quantized_url = OpenSearch::Quantize.format_url(url)
+                  span.resource = "#{method} #{quantized_url}"
+                rescue StandardError => e
+                  Datadog.logger.error(e.message)
+                ensure
+                  begin
+                    response = super
+                  rescue => e
+                    status_code = ::OpenSearch::Transport::Transport::ERRORS.key(e.class)
+                    span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_STATUS_CODE, status_code) if status_code
+                    raise
+                  end
+                  # Set post-response tags
+                  span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_STATUS_CODE, response.status)
+                  if response.headers['content-length']
+                    span.set_tag(
+                      OpenSearch::Ext::TAG_RESPONSE_CONTENT_LENGTH,
+                      response.headers['content-length'].to_i
+                    )
+                  end
+                end
+              end
+              response
+            end
+
+            def tag_params(params, span)
+              if params
+                params = JSON.generate(params) unless params.is_a?(String)
+                span.set_tag(OpenSearch::Ext::TAG_PARAMS, params)
+              end
+            end
+
+            def tag_body(body, span)
+              if body
```

Comment:
Minor style: You could put `if body` as `return unless body` and remove the indentation here. Same applies to `tag_params`.